### PR TITLE
[ONNX] Update Default Value of recompute_scale_factor in Interpolate

### PR DESCRIPTION
**Pull request by neginraoof**
This is a duplicate of https://github.com/pytorch/pytorch/pull/38362

"This PR completes Interpolate's deprecation process for recomputing the scales values, by updating the default value of the parameter recompute_scale_factor as planned for pytorch 1.6.0.
The warning message is also updated accordingly."

I'm recreating this PR as previous one is not being updated.

cc @gchanan

### CODE REVIEW

**Comment by neginraoof** [comment]
@gchanan This is a duplicate of https://github.com/pytorch/pytorch/pull/38362 as the old PR is not being updated.
cc @houseroad @BowenBao 
Could you please help review this PR?

**Comment by neginraoof** [comment]
This is a duplicate of this PR: https://github.com/pytorch/pytorch/pull/38362

cc @gchanan for review. Thanks.

**Comment by neginraoof** [comment]
@houseroad @gchanan Would you please help review this PR? Thanks

**Comment by neginraoof** [comment]
This PR is pending model accuracy tests using torchvision models.
cc @houseroad @fmassa

**Comment by fmassa** [comment]
@mthrok did accuracy testing on `keypoint_rcnn_resnet50_fpn`, and saw no sign of accuracy drop by switching to `recompute_scale_factor=False`.

**Comment by neginraoof** [comment]
cc @houseroad for review.
Thanks

**Review by houseroad — COMMENTED** [review]
Before landing, could you rebase to the master and trigger the test again?

**Comment by neginraoof** [comment]
@houseroad This PR is rebased. Thanks.

**Comment by neginraoof** [comment]
cc @houseroad for review. Thanks.

**Comment by neginraoof** [comment]
@houseroad Would you please let us know if there's an internal failure or the PR is ready?

Thanks!

**Comment by neginraoof** [comment]
@houseroad could you please let us know about internal failures and if there's anything I can do to land this?
Thanks!

**Review by houseroad — COMMENTED** [review]
Has confirmed with @fmassa @ppwwyyxx, this PR should not impact torch vision and detectron2.

**Comment by houseroad** [comment]
I tried to land the PR, however it's conflicting with the trunk. Could you rebase the PR?

**Comment by mthrok** [comment]
> @mthrok did accuracy testing on `keypoint_rcnn_resnet50_fpn`, and saw no sign of accuracy drop by switching to `recompute_scale_factor=False`.

Note here https://fb.quip.com/8cYGADPqDJL5

File: torch/csrc/api/include/torch/nn/functional/upsampling.h [hunk 1]
```diff
@@ -98,7 +98,7 @@ inline Tensor interpolate(
 
   auto scale_factor_len = input.dim() - 2;
   std::vector<c10::optional<double>> scale_factor_list(scale_factor_len, c10::nullopt);
-  if (scale_factor != c10::nullopt && recompute_scale_factor.has_value() && !recompute_scale_factor.value()) {
+  if (scale_factor != c10::nullopt && (recompute_scale_factor == c10::nullopt || !recompute_scale_factor.value_or(false))) {
```
Comment (houseroad) [inline torch/csrc/api/include/torch/nn/functional/upsampling.h:101]: Nit: (recompute_scale_factor == c10::nullopt || !recompute_scale_factor.value_or(false)) => !recompute_scale_factor.value_or(false)

**Review by houseroad — APPROVED** [review]

File: torch/nn/functional.py [hunk 2]
```diff
@@ -3055,8 +3054,9 @@ def interpolate(input, size=None, scale_factor=None, mode='nearest', align_corne
 
     scale_factor_len = input.dim() - 2
     scale_factor_list = torch.jit.annotate(List[Optional[float]], [None for _ in range(scale_factor_len)])
-    if scale_factor is not None and recompute_scale_factor is False:
-        if isinstance(scale_factor, list):
+    # default value of recompute_scale_factor is False
+    if scale_factor is not None and (recompute_scale_factor is False or recompute_scale_factor is None):
+        if isinstance(scale_factor, list) or isinstance(scale_factor, tuple):
```
Comment (gchanan) [inline torch/nn/functional.py:3059]: nit: `isinstance(scale_factor, (list, tuple))`

**Pull request closed without merging**